src/Microsoft.VisualStudio.Web.CodeGeneration.MsBuild/ProjectUtilities.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.VisualStudio.Web.CodeGeneration.ProjectInfo;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.MsBuild
+{
+    public class ProjectUtilities
+    {
+        private const string ItemSpecKey = "OriginalItemSpec";
+        private const string NugetSourceTypeKey = "NugetSourceType";
+        private const string VersionKey = "Version";
+
+        public static Microsoft.Build.Evaluation.Project CreateProject(string filePath, string configuration, IDictionary<string, string> globalProperties)
+        {
+            var xmlReader = XmlReader.Create(new FileStream(filePath, FileMode.Open));
```

Comment:
Recommend closing the stream once you are done with it. Otherwise your process may hold a lock on the csproj file